native/arrow_format_nif/src/message/header.rs

```diff
@@ -0,0 +1,29 @@
+use super::record_batch::RecordBatch;
+use super::schema::Schema;
+use rustler::{Decoder, Encoder, Term};
+
+#[derive(Debug)]
+pub enum Header {
+    Schema(Schema),
+    RecordBatch(RecordBatch),
+}
```

Comment:
As well as for this

Reply:
Actually, [`NifUntaggedEnum`](https://docs.rs/rustler_codegen/0.29.1/src/rustler_codegen/lib.rs.html#360) worked best with this.